File: day1/main.js
const fs = require("fs");
const readline = require("readline");

const input_path = "input.csv";

const inputStream = fs.createReadStream(input_path);
lst1 = [];
lst2 = [];
async function processLineByLine() {
  const rl = readline.createInterface({
    input: inputStream,
      terminal: false,
  });

  for await (const line of rl) {
    output = line.split("   ");
    lst1.push(output[0]);
    lst2.push(output[1]);
  }
};

processLineByLine().then(()=>{
  lst1.sort();
  lst2.sort();
  sum = 0;
  for(let i = 0; i< lst1.length; i++){
    sum += Math.abs(lst1[i] - lst2[i]);

  }
  console.log(sum);

  count_sum = 0;
  for(let i=0; i < lst1.length; i++){
    count_sum += lst1[i] * lst2.filter(x => x==lst1[i]).length;
  }


  console.log(count_sum);
});

File: day1/main.py

lst1 = []
lst2 = []

with open("input.csv", "r") as f:
  for line in f.readlines():
    input1, input2 = line.split()
    lst1.append(int(input1))
    lst2.append(int(input2))
  

lst1_sorted = sorted(lst1)
lst2_sorted =sorted(lst2)


sum = 0

for i in range(len(lst1)):
  sum += abs(lst1_sorted[i] - lst2_sorted[i])
  #print(lst1[i], lst2[i], abs(lst1[i] - lst2[i]))
print(sum)

count_sum = 0
for i in range(len(lst1_sorted)):
  count_sum += lst1_sorted[i] * lst2_sorted.count(lst1_sorted[i])

print(count_sum)
